sycl/doc/extensions/supported/sycl_ext_oneapi_bfloat16.asciidoc

```diff
@@ -22,69 +22,59 @@
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 5.
 
-Because the interfaces defined by this specification are not final and are
-subject to change they are not intended to be used by shipping software
-products.
+== Status
 
-== Version
+This extension is implemented and fully supported by DPC++.
 
-Revision: 5
 
-== Introduction
+== Overview
 
-This extension adds functionality to convert value of single-precision
-floating-point type(`float`) to `bfloat16` type and vice versa. The extension
-doesn't add support for `bfloat16` type as such, instead it uses 16-bit integer
-type(`uint16_t`) as a storage for `bfloat16` values.
+This extension adds support for a 16-bit floating point type `bfloat16`. This type occupies 16 bits of storage space as does the `sycl::half` type. However, `bfloat16` allots 8 bits to the exponent instead of the 5 bits used by `sycl::half` and 7 bits to the significand versus 10 bits used by `sycl::half`. Thus, `bfloat16` has the same dynamic range as a 32-bit `float` but with reduced precision. This type is useful when memory required to store the values must be reduced, and when the calculations require high dynamic range but can tolerate lower-precision. Some implementations may still perform operations on this type using 32-bit math. For example, they may convert the `bfloat16` value to `float`, and then perform the operation on the 32-bit `float`.
 
-The purpose of conversion from float to bfloat16 is to reduce the amount of memory
-required to store floating-point numbers. Computations are expected to be done with
-32-bit floating-point values.
 
-This extension is an optional kernel feature as described in
-https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:optional-kernel-features[section 5.7]
-of the SYCL 2020 spec. Therefore, attempting to submit a kernel using this
-feature to a device that does not support it should cause a synchronous
-`errc::kernel_not_supported` exception to be thrown from the kernel invocation
-command (e.g. from `parallel_for`).
+== Specification
 
-== Feature test macro
+=== Feature test macro
 
 This extension provides a feature-test macro as described in the core SYCL
 specification section 6.3.3 "Feature test macros". Therefore, an implementation
 supporting this extension must predefine the macro
 `SYCL_EXT_ONEAPI_BFLOAT16` to one of the values defined in the table
 below. Applications can test for the existence of this macro to determine if
 the implementation supports this feature, or applications can test the macro’s
- value to determine which of the extension’s APIs the implementation supports.
+ value to determine which of the extension’s APIs the implementation supports. Attempting to submit a kernel using `bfloat16` to a device that does not support it causes a synchronous `errc::kernel_not_supported` exception to be thrown from the kernel invocation command (e.g. from `parallel_for`).
```

Comment:
I think this sentence is no longer correct:

> Attempting to submit a kernel using `bfloat16` to a device that does not support it causes a synchronous `errc::kernel_not_supported` exception to be thrown from the kernel invocation command (e.g. from `parallel_for`).

I think you added fall-back support, correct?  If so, there is no error when submitting a `bfloat16` kernel that does not have native support and there is no exception.  The NOTE below is also irrelevant in that case.

Reply:
This is intended to cover the case where the user has not used the macro SYCL_EXT_ONEAPI_BFLOAT16 to check if the feature is available and uses bfloat16 in an environment where bfloat16 support does not exist.

Reply:
The only time the macro `SYCL_EXT_ONEAPI_BFLOAT16` is unset is when the implementation doesn't support this extension.  In that case, the type `bfloat16` isn't even defined, so you will get a compilation error.

Reply:
OK, removed the text about runtime error from kernel.